qiskit/visualization/gate_map.py

```diff
@@ -1051,128 +1033,86 @@ def plot_coupling_map(
-            x_start = grid_data[edge[0]][1]
-            y_end = grid_data[edge[1]][0]
-            x_end = grid_data[edge[1]][1]
-
-            if is_symmetric:
-                if y_start == y_end:
-                    x_end = (x_end - x_start) / 2 + x_start
-
-                elif x_start == x_end:
-                    y_end = (y_end - y_start) / 2 + y_start
+        line_color = ["#648fff"] * len(coupling_map)
 
-                else:
-                    x_end = (x_end - x_start) / 2 + x_start
-                    y_end = (y_end - y_start) / 2 + y_start
-            ax.add_artist(
-                plt.Line2D(
-                    [x_start, x_end],
-                    [-y_start, -y_end],
-                    color=line_color[ind],
-                    linewidth=line_width,
-                    zorder=0,
-                )
-            )
-            if plot_directed:
-                dx = x_end - x_start
-                dy = y_end - y_start
-                if is_symmetric:
-                    x_arrow = x_start + dx * 0.95
-                    y_arrow = -y_start - dy * 0.95
-                    dx_arrow = dx * 0.01
-                    dy_arrow = -dy * 0.01
-                    head_width = 0.15
-                else:
-                    x_arrow = x_start + dx * 0.5
-                    y_arrow = -y_start - dy * 0.5
-                    dx_arrow = dx * 0.2
-                    dy_arrow = -dy * 0.2
-                    head_width = 0.2
-                ax.add_patch(
-                    mpatches.FancyArrow(
-                        x_arrow,
-                        y_arrow,
-                        dx_arrow,
-                        dy_arrow,
-                        head_width=head_width,
-                        length_includes_head=True,
-                        edgecolor=None,
-                        linewidth=0,
-                        facecolor=line_color[ind],
-                        zorder=1,
-                    )
-                )
-
-    # Add circles for qubits
-    for var, idx in enumerate(grid_data):
-        _idx = [idx[1], -idx[0]]
-        ax.add_artist(
-            mpatches.Ellipse(
-                _idx,
-                qubit_size / 48,
-                qubit_size / 48,  # This is here so that the changes
-                color=qubit_color[var],
-                zorder=1,
-            )
-        )  # to how qubits are plotted does
-        if label_qubits:  # not affect qubit size kwarg.
-            ax.text(
-                *_idx,
-                s=qubit_labels[var],
-                horizontalalignment="center",
-                verticalalignment="center",
-                color=font_color,
-                size=font_size,
-                weight="bold",
-            )
-    ax.set_xlim([-1, x_max + 1])
-    ax.set_ylim([-(y_max + 1), 1])
-    ax.set_aspect("equal")
+    graph = CouplingMap(coupling_map).graph
+
+    if not plot_directed:
+        graph = graph.to_undirected()
```

Comment:
The image comparison tests caught this edge case, we need to flatten parallel edges here with:

```suggestion
        graph = graph.to_undirected(multigraph=False)
```

see: https://qiskit.org/documentation/retworkx/apiref/rustworkx.PyDiGraph.to_undirected.html

Otherwise graphs that should look like:

![65_qubit_gate_map](https://github.com/Qiskit/qiskit-terra/assets/2447371/97f6328a-8b34-419b-92d2-ab7891403174)

end up looking like:

![65_qubit_gate_map](https://github.com/Qiskit/qiskit-terra/assets/2447371/96341fa8-9bfd-4670-a4cc-85a764e9e405)